analyses/ssd/flood_trigger/ssd_floodscan.md

```diff
@@ -0,0 +1,372 @@
+A specific focus is on a subset of the counties in the states of Jonglei and Unity. 
+As these experience flooding and might be the focus of the pilot
+
+```python
+import os
+from pathlib import Path
+import sys
+from datetime import timedelta
+import xarray as xr
+import matplotlib.pyplot as plt
+import seaborn as sns
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+import geopandas as gpd
+import hvplot.xarray
+import matplotlib as mpl
+import numpy as np
+from scipy import stats
+from functools import reduce
+import altair as alt
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+
+from src.indicators.drought.config import Config
+from src.indicators.flooding.floodscan import floodscan
+from src.utils_general.raster_manipulation import compute_raster_statistics
+
+
+mpl.rcParams['figure.dpi'] = 300
+```
+
+```python
+iso3="ssd"
+config=Config()
+parameters = config.parameters(iso3)
+country_data_raw_dir = Path(config.DATA_DIR) / config.PUBLIC_DIR / config.RAW_DIR / iso3
+country_data_exploration_dir = Path(config.DATA_DIR) / config.PRIVATE_DIR / "exploration" / iso3
+adm2_bound_path=country_data_raw_dir / config.SHAPEFILE_DIR / parameters["path_admin2_shp"]
+```
+
+```python
+#admin2's of interest for now
+#all located in Unity and Jonglei
+adm2_list=['Panyijiar', 'Leer', 'Mayendit', 'Koch', 'Guit',
+           'Fangak', 'Ayod', 'Duk', 'Twic East', 'Bor South']
+```
+
+```python
+gdf_adm2=gpd.read_file(adm2_bound_path)
+```
+
+```python
+# #clip to country
+# #this takes long, so only do when not saved the file yet
+# #read floodscan data
+# fs = floodscan.Floodscan()
+# fs_raw = fs.read_raw_dataset()
+# fs_clip = (fs_raw.rio.set_spatial_dims(x_dim="lon", y_dim="lat")
+#            .rio.write_crs("EPSG:4326")
+#            .rio.clip(gdf_adm2["geometry"], all_touched=True))
+```
+
+```python
+# #somehow cannot save the file with these attributes
+# fs_clip.SFED_AREA.attrs.pop('grid_mapping')
+# fs_clip.NDT_SFED_AREA.attrs.pop('grid_mapping')
+# fs_clip.LWMASK_AREA.attrs.pop('grid_mapping')
+# fs_clip.to_netcdf(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan.nc')
+```
+
+```python
+fs_clip=xr.load_dataset(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan.nc')
+#I dont fully understand why, these grid mappings re-occur and what they mean
+#but if having them, later on getting
+fs_clip.SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.NDT_SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.LWMASK_AREA.attrs.pop('grid_mapping')
+fs_clip=fs_clip.rio.write_crs("EPSG:4326",inplace=True)
+```
+
+We plot the data. We can see that the resolution is high (300 arcseconds = 0.083 degrees). We therefore decide that when analyzing the data at the admin2 level, it is fine to only look at the cells with their centre within the admin2. 
+
+```python
+g=fs_clip.sel(time='2020-08-03T00:00:00.000000000').SFED_AREA.plot()
+gdf_adm2.boundary.plot(ax=g.axes);
```

Comment:
Btw here I was playing around a bit with `hvplot` and `groupby='time'` to get a scroll bar. I couldn't quite get it to work though.